tools/validators/instance_validator/universe_generation/generate_universe.py

```diff
@@ -0,0 +1,213 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Generates the ontology universe for the instance validator."""
+
+from __future__ import print_function
+from __future__ import absolute_import
+
+from os import path
+
+from yamlformat.validator import external_file_lib
+from yamlformat.validator import presubmit_validate_types_lib
+from yamlformat.validator import namespace_validator
+from yamlformat.validator import state_lib
+from yamlformat.validator import subfield_lib
+from yamlformat.validator import unit_lib
+from yamlformat.validator import entity_type_lib
+from yamlformat.validator import field_lib
+from yamlformat.validator import parse_config_lib as parse
+
+def build_config():
+  """Builds and returns config for use in generating universes.
+
+  Returns:
+    config: a Config namedtuple containing lists of localpaths to config files.
+  """
+
+  yaml_files = external_file_lib._RecursiveDirWalk(path.join(
+      '..', '..', '..', 'ontology', 'yaml', 'resources'))
+  config = presubmit_validate_types_lib.SeparateConfigFiles(yaml_files)
+
+  return config
+
+def build_state_universe(config):
```

Comment:
why not have one function that takes config and type and returns state or unit or ...?

Reply:
Some of the sub-universes depend on each other to build, such as `UnitUniverse` depending on `SubfieldUniverse`. So if I have one function that can return any of the sub-universes, my feeling was that it will end up similar to the existing code where I generate the entire universe?

Reply:
I don't see any reason NOT to generate all the universes together.  Breaking them up is only going to be more error prone.  The correct behavior is to build the entire set of universes, then extract anything you need directly from the config universe.  You can then add functions to the config universe to help you extract elements you need.

Reply:
Done